Sources/MongoSwift/APM.swift

```diff
@@ -117,6 +117,90 @@ private protocol CommandEventProtocol {
     var serviceID: BSONObjectID? { get }
 }
 
+#if compiler(>=5.5.2) && canImport(_Concurrency)
+/// An asynchronous way to monitor events that uses `AsyncSequence`.
+/// Only available for Swift 5.5.2 and higher.
+@available(macOS 10.15, *)
+// sourcery: skipSyncExport
+public struct EventStream<T> {
+    private var stream: AsyncStream<T>
+    private var continuation: AsyncStream<T>.Continuation?
```

Comment:
since the handlers already store the continuation do we need to store it here too?

Reply:
If the `.finish()` method is re-assigned to the handler as per previous comments, then no, it seems like we don't need it.